src/main/java/com/limechain/sync/warpsync/state/RequestFragmentsState.java

```diff
@@ -47,15 +45,25 @@ public void next(WarpSyncMachine sync) {
 
     @Override
     public void handle(WarpSyncMachine sync) {
+        WarpSyncResponse resp = null;
+        for (int i = 0; i < sync.getNetworkService().kademliaService.getBootNodePeerIds().size(); i++) {
+            try {
+                resp = sync.getNetworkService().makeWarpSyncRequest(blockHash.toString());
+                break;
+            } catch (Exception e) {
+                if (!sync.getNetworkService().updateCurrentSelectedPeerWithBootnode(i)) {
+                    this.error = new RuntimeException("Could not update current selected peer");
+                    return;
+                }
+            }
```

Comment:
It's always good to print the initial exception (in this case `e`) because sometimes it's different than what we expect.